### PR TITLE
Fix navtray mobile links

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch7833]

File: src/features/Layout/Navigation/TopNav.tsx [hunk 1]
```diff
@@ -62,14 +62,14 @@ export const TopNav = ({
...
-        current={locked}
+        current={!isTrayOpen && locked}
         color={color}
       />
       <TopItem
         title="NAVIGATION_JOIN"
         icon="nav-membership"
         onClick={() => push(ROUTE_PATHS.MYC_MEMBERSHIP.path)}
-        current={current === ROUTE_PATHS.MYC_MEMBERSHIP.path}
+        current={!isTrayOpen && current === ROUTE_PATHS.MYC_MEMBERSHIP.path}
```
Comment (GuillaumeRx) [inline src/features/Layout/Navigation/TopNav.tsx:72]: On desktop this blocks the current item to show

File: src/features/Layout/Navigation/components/NavTray.tsx [hunk 2]
```diff
@@ -24,15 +25,26 @@ const SIcon = styled(Icon)`
...
+  const handleToggle = () => {
+    if (!isMobile) return;
+    setOpen(!isOpen);
+  };
+
+  useClickAway(ref, () => isMobile && setOpen(false));
+
   const handleOpen = () => {
+    if (isMobile) return;
     isReady() === false && clear();
```
Comment (GuillaumeRx) [inline src/features/Layout/Navigation/components/NavTray.tsx:43]: On mobile this already worked without the need of a `useClickAway`

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: src/features/Layout/Navigation/TopNav.tsx (hunk 1 above)
Comment (FrederikBolding) [reply src/features/Layout/Navigation/TopNav.tsx:72]: Will fix 👍

File: src/features/Layout/Navigation/components/NavTray.tsx (hunk 2 above)
Comment (FrederikBolding) [reply src/features/Layout/Navigation/components/NavTray.tsx:43]: This way it closes when you hit a tray link 👍

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**